### PR TITLE
Improves test cases

### LINKED ISSUE #264: Improve JUnit tests for undo and redo feature

### PR DESCRIPTION
**Pull request by lightz96**
1. Shifts the initialization of `PastUndoableCommands` within the constructor of `ModelManager`.
2. Adds more tests for undoable commands.
3. Adds assertion. 

Resolves #264

### CODE REVIEW

File: src/test/java/thrift/logic/commands/AddExpenseCommandIntegrationTest.java [hunk 1]
```diff
@@ -24,60 +25,41 @@
...
         expectedModel.addExpense(validExpense);
 
         assertCommandSuccess(new AddExpenseCommand(validExpense), model,
                 String.format(AddExpenseCommand.MESSAGE_SUCCESS, validExpense), expectedModel);
     }
 
     @Test
-    public void undo_undoAddExpense_success() throws CommandException {
-        Model expectedModel = new ModelManager(model.getThrift(), new UserPrefs(), new PastUndoableCommands());
+    public void undoAndRedo_undoAndRedoAddExpense_success() throws CommandException {
```
Comment (krusagiz) [inline src/test/java/thrift/logic/commands/AddExpenseCommandIntegrationTest.java]: Will `undoAndRedo_AddExpense_success()` look better?

File: src/test/java/thrift/logic/commands/AddIncomeCommandIntegrationTest.java [hunk 2]
```diff
@@ -24,60 +25,40 @@
...
         expectedModel.addIncome(validIncome);
 
         assertCommandSuccess(new AddIncomeCommand(validIncome), model,
                 String.format(AddIncomeCommand.MESSAGE_SUCCESS, validIncome), expectedModel);
     }
 
     @Test
-    public void undo_success() throws CommandException {
-        Model expectedModel = new ModelManager(model.getThrift(), new UserPrefs(), new PastUndoableCommands());
+    public void undoAndRedo_undoAndRedoAddIncome_success() throws CommandException {
```
Comment (krusagiz) [inline src/test/java/thrift/logic/commands/AddIncomeCommandIntegrationTest.java]: Will `undoAndRedo_AddIncome_success()` look better?

File: src/test/java/thrift/logic/commands/CloneCommandTest.java [hunk 3]
```diff
@@ -202,122 +207,205 @@ public void execute_invalidIndexValidOccurrenceFilteredList_throwsCommandExcepti
     }
 
     @Test
-    public void undo_undoCloneCommandExpenseWithNoOccurrence_success() {
-        Model expectedModel = new ModelManager(model.getThrift(), new UserPrefs(), new PastUndoableCommands());
+    public void undoAndRedo_undoAndRedoCloneCommandOnExpenseWithNoOccurrence_success() {
```
Comment (krusagiz) [inline src/test/java/thrift/logic/commands/CloneCommandTest.java]: Will `undoAndRedo_CloneCommandOnExpenseWithNoOccurrence_success()` look better?

File: src/test/java/thrift/logic/commands/DeleteCommandTest.java [hunk 4]
```diff
@@ -106,88 +102,47 @@ public void equals() {
     }
 
     @Test
-    public void undo_undoDeleteExpense_success() throws CommandException {
-        Model expectedModel = new ModelManager(model.getThrift(), new UserPrefs(),
-                new PastUndoableCommands());
+    public void undoAndRedo_undoAndRedoDeleteExpense_success() throws CommandException {
```
Comment (krusagiz) [inline src/test/java/thrift/logic/commands/DeleteCommandTest.java]: Will `undoAndRedo_DeleteExpense_success()` look better?

File: src/test/java/thrift/logic/commands/TagCommandTest.java [hunk 5]
```diff
@@ -56,74 +60,73 @@ void execute_newTags_success() throws CommandException {
...
     @Test
-    void undo_undoTag_success() {
-        Model model = new ModelManager(TypicalTransactions.getTypicalThrift(), new UserPrefs(),
-                new PastUndoableCommands());
-        Model expectedModel = new ModelManager(model.getThrift(), new UserPrefs(),
-                new PastUndoableCommands());
-
-        Index indexLastTransaction = Index.fromOneBased(model.getFilteredTransactionList().size());
-        Transaction lastTransaction = model.getFilteredTransactionList().get(indexLastTransaction.getZeroBased());
+    void undoAndRedo_undoAndRedoTagForExpense_success() {
```
Comment (krusagiz) [inline src/test/java/thrift/logic/commands/TagCommandTest.java]: Will `undoAndRedo_TagForExpense_success()` look better?

**Review by krusagiz — CHANGES_REQUESTED** [review]
LGTM overall. Should be fine after some test case names are changed, I have commented on a few of them as a suggestion but more should be changed. 

Try to make it in this format: `(method(s))_(conditions)_(success|failure)`.

File: src/test/java/thrift/logic/commands/AddExpenseCommandIntegrationTest.java (hunk 1 above)
Comment (lightz96) [reply src/test/java/thrift/logic/commands/AddExpenseCommandIntegrationTest.java]: Fixed

File: src/test/java/thrift/logic/commands/CloneCommandTest.java (hunk 3 above)
Comment (lightz96) [reply src/test/java/thrift/logic/commands/CloneCommandTest.java]: Fixed

File: src/test/java/thrift/logic/commands/TagCommandTest.java (hunk 5 above)
Comment (lightz96) [reply src/test/java/thrift/logic/commands/TagCommandTest.java]: Fixed

File: src/test/java/thrift/logic/commands/DeleteCommandTest.java (hunk 4 above)
Comment (lightz96) [reply src/test/java/thrift/logic/commands/DeleteCommandTest.java]: Fixed

File: src/test/java/thrift/logic/commands/AddIncomeCommandIntegrationTest.java (hunk 2 above)
Comment (lightz96) [reply src/test/java/thrift/logic/commands/AddIncomeCommandIntegrationTest.java]: Fixed

**Review by krusagiz — APPROVED** [review]
LGTM!

**Pull request merged**